### ISSUE TITLE
Collection: An item with the same key has already been added 

### ISSUE THREAD

**Issue by DevEddy** [markmeeus/MarcelloDB#74]
Hi!

sometimes I get this error. Do you have any idea why this could happen?

```chsarp
An item with the same key has already been added. Key: Displays Stack:   at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x0015a] in <b62d44c4e9274f7e95d2feb83edd0883>:0 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <b62d44c4e9274f7e95d2feb83edd0883>:0 
  at MarcelloDB.Collections.CollectionFile+<>c__DisplayClass13_0`2[T,TID].<Collection>b__0 () [0x00077] in <ce43c20e305646cf958407e87464f571>:0 
  at MarcelloDB.Session.Transaction (System.Action action) [0x00053] in <ce43c20e305646cf958407e87464f571>:0 
  at MarcelloDB.Collections.CollectionFile.Collection[T,TID] (System.String collectionName, System.Func`2[T,TResult] idFunc) [0x00064] in <ce43c20e305646cf958407e87464f571>:0 
  at CloudMatic.DAL.MarcelloDatabase.MarcelloDatabase.GetCollection[T] (System.String additionalInfoTocollectionName) [0x00023] in <b628bd5e75dd4fd49c52026d3a881b84>:0 
```

Edit: I tried with lock and it works as expected. I think it has something to do with parallel tasks. You can close the issue if you think it's a user problem.

Thank you,
Eddy

**Comment by markmeeus** [comment]
HI Eddy,

This is definitely not an user problem, MarcelloDB should protect itself from threading issues. 
I'll look into it.

**Comment by markmeeus** [comment]
I have just released a prelease package, 1.0.9-beta on nuget.org
I don't know on which platforms you are running, if it's Android or iOS you should be fine, I have not recompiled the uwp assembly against 1.0.9 in this beta version, so let me know if you need to test that one.

Could you check if this solves your issues?

**Comment by DevEddy** [comment]
Hi Mark,

I am using Marcello on iOS, Android and UWP in this project.

Sure I can test. I will let you know.

**Comment by DevEddy** [comment]
I've tested it on iOS and Android with 1.0.9-beta. Everything works now as expected without the lock.

**Comment by markmeeus** [comment]
Thank you for testing, I have just released version 1.0.9 on nuget, should be available soon.
